src/gwt/src/org/rstudio/studio/client/workbench/views/source/NewShinyWebApplication.java

```diff
@@ -0,0 +1,172 @@
+package org.rstudio.studio.client.workbench.views.source;
+
+import org.rstudio.core.client.widget.DirectoryChooserTextBox;
+import org.rstudio.core.client.widget.ModalDialog;
+import org.rstudio.core.client.widget.OperationWithInput;
+import org.rstudio.core.client.widget.SelectWidget;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.core.client.Scheduler;
+import com.google.gwt.core.client.Scheduler.ScheduledCommand;
+import com.google.gwt.dom.client.Style.Unit;
+import com.google.gwt.resources.client.ClientBundle;
+import com.google.gwt.resources.client.CssResource;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.Grid;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+public class NewShinyWebApplication extends ModalDialog<NewShinyWebApplication.Result>
+{
+   public static class Result
+   {
+      public Result(String appName, String appType, String appDir)
+      {
+         appName_ = appName;
+         appType_ = appType;
+         appDir_ = appDir;
+      }
+      
+      public String getAppName() { return appName_; }
+      public String getAppType() { return appType_; }
+      public String getAppDir() { return appDir_; }
+      
+      private String appName_;
+      private String appType_;
+      private String appDir_;
+   }
+   
+   private static class VerticalSpacer extends Composite
+   {
+      public VerticalSpacer(String height)
+      {
+         FlowPanel panel = new FlowPanel();
+         panel.setHeight(height);
+         initWidget(panel);
+      }
+   }
+   
+   public NewShinyWebApplication(String caption, 
+                                 String workingDirectory,
+                                 OperationWithInput<Result> operation)
+   {
+      super(caption, operation);
+      
+      container_ = new VerticalPanel();
+      
+      // Create individual widgets
+      appNameLabel_ = new Label("Application Name:");
+      appNameLabel_.addStyleName(RES.styles().label());
+      
+      appNameTextBox_ = new TextBox();
+      appNameTextBox_.getElement().getStyle().setMarginLeft(10, Unit.PX);
+      appNameTextBox_.getElement().setAttribute("placeholder", "Name");
+      appNameTextBox_.setText(defaultAppName(workingDirectory));
```

Comment:
Because we use the directory chooser widget, users can't manually enter paths (instead they have to use the breadcrumb widget). Is that sufficient here?